src/SDK/Common/Retry/RetryPolicyInterface.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Retry;
+
+// use Opentelemetry\Proto\Collector\Trace\V1\ExportTraceServiceResponse;
+use OpenTelemetry\SDK\Metrics\Exceptions\RetryableExportException;
+
+interface RetryPolicyInterface
+{
+    public const DEFAULT_MAX_ATTEMPTS = 5;
+    public const DEFAULT_INITIAL_BACKOFF = 1;
+    public const DEFAULT_MAX_BACKOFF = 10;
+    public const DEFAULT_BACKOFF_MULTIPLIER = 1.5;
+    public const DEFAULT_JITTER = 0.1;
+
+    /**
+     * Returns whether the request should be retried.
+     *
+     * @param int $attempt - current number of retry attempt
+     * @param int $status - SpanExporterStatusInterface
+     * @param RetryableExportException|null $exception: exception caught while retrying
+     *
+     * @return bool
+     */
+    public function shouldRetry(
+        int $attempt,
+        int $status,
+        ?RetryableExportException $exception
+    ): ?bool;
+
+    /**
+     * Returns the time to wait in seconds.
+     * @param int $attempt: current number of retry attempt
+     *
+     * @return int
+     */
+    public function getDelay(int $attempt): int;
+
+    public static function getDefault(): RetryPolicyInterface;
+
+    public function getMaxAttempts(): int;
+
+    public function setMaxAttempts(int $max_attempts);
+
+    public function getInitialBackoff(): float;
+
+    public function setInitialBackoff(float $initial_backoff);
+
+    public function getMaxBackoff(): int;
+
+    public function setMaxBackoff(int $max_backoff);
+
+    public function getBackoffMultiplier(): float;
+
+    public function setBackoffMultipler(float $backoff_multiplier);
```

Comment:
Should remove (most) of the getters/setters as these values are specific to the exponential retry policy and not relevant to retry strategies in general.

Reply:
As per the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#retry), it is mandatory to implement exponential retry with jitter in otel exporter. That is the reason I have put all these values in RetryPolicyInterface.php and not in ExponentialRetry.php

Reply:
I'm still in favor of removing the getters/setters to have less mutable state.
Additionally other exponential retry policies might not support all of these configurations.